modules/alloy/src/main/ts/ephox/alloy/positioning/mode/ContentAnchorCommon.ts

```diff
@@ -9,8 +9,11 @@ import * as Origins from '../layout/Origins';
 import { Anchoring, NodeAnchor, nu as NuAnchor, SelectionAnchor } from './Anchoring';
 import * as AnchorLayouts from './AnchorLayouts';
 
-const capRect = (left: number, top: number, width: number, height: number): Optional<Boxes.BoxByPoint> => {
-  let newLeft = left, newTop = top, newWidth = width, newHeight = height;
+const capRect = (left: number, top: number, width: number, height: number, cappingViewport: Optional<Boxes.Bounds>): Optional<Boxes.BoxByPoint> => {
+  let newLeft = left;
+  let newTop = top;
+  let newWidth = width;
+  let newHeight = height;
   // Try to prevent the context toolbar from getting above the editor toolbar
   if (left < 0) {
```

Comment:
Are you intentionally capping to cap.width and cap.height for the bottom, but 0 and 0 for the top left? Should it be cap.left or cap.top or cap.x or cap.y ? And if not, should cappingViewport just be a width/height value rather than a rectangle to avoid confusion?

Reply:
The square is provided as relative, and is later put into where exactly the view is. So the top can come as a negative value, which is then how far it is out of the view. So when the code caps it at 0, it caps it at the top of the viewable area.
I'm not sure why this was implemented this way, but it appears to be so as far as I could tell.